src/views/domain-workflows-basic/domain-workflows-basic-table/__tests__/domain-workflows-basic-table.test.tsx

```diff
@@ -0,0 +1,280 @@
+    expect(screen.getByText(`mock-workflow-id-0-4-closed`)).toBeInTheDocument();
+    expect(screen.getByText(`mock-workflow-id-0-5-closed`)).toBeInTheDocument();
+    expect(screen.getByText(`mock-workflow-id-0-6-closed`)).toBeInTheDocument();
+    expect(screen.getByText(`mock-workflow-id-0-7-closed`)).toBeInTheDocument();
+    expect(screen.getByText(`mock-workflow-id-0-8-closed`)).toBeInTheDocument();
+    expect(screen.getByText(`mock-workflow-id-0-9-closed`)).toBeInTheDocument();
+  });
+});
+
+function setup({
+  errorCase,
+}: {
+  errorCase?: 'initial-fetch-error' | 'subsequent-fetch-error' | 'no-workflows';
+}) {
+  const openPages = generateWorkflowPages(2, true);
+  const closedPages = generateWorkflowPages(2);
+
+  let currentEventIndexOpen = 0;
+  let currentEventIndexClosed = 0;
+  const user = userEvent.setup();
+
+  const renderResult = render(
+    <DomainWorkflowsBasicTable domain="mock-domain" cluster="mock-cluster" />,
+    {
+      endpointsMocks: [
+        {
+          path: '/api/domains/:domain/:cluster/workflows-basic',
+          httpMethod: 'GET',
+          mockOnce: false,
+          httpResolver: async ({ request }) => {
+            const url = new URL(request.url);
+            const kind = url.searchParams.get('kind');
+
+            let index;
+            if (kind === 'closed') {
+              index = currentEventIndexClosed;
+              currentEventIndexClosed++;
+            } else {
+              index = currentEventIndexOpen;
+              currentEventIndexOpen++;
+            }
+
+            const pages = kind === 'closed' ? closedPages : openPages;
+
+            switch (errorCase) {
+              case 'no-workflows':
+                return HttpResponse.json({
+                  workflows: [],
+                  nextPage: undefined,
+                });
+              case 'initial-fetch-error':
+                return HttpResponse.json(
+                  { message: 'Request failed' },
+                  { status: 500 }
+                );
+              case 'subsequent-fetch-error':
+                if (index === 0) {
+                  return HttpResponse.json(pages[0]);
+                } else if (index === 1) {
+                  return HttpResponse.json(
+                    { message: 'Request failed' },
+                    { status: 500 }
+                  );
+                } else {
+                  return HttpResponse.json(pages[1]);
+                }
+              default:
+                if (index === 0) {
+                  return HttpResponse.json(pages[0]);
+                } else {
+                  return HttpResponse.json(pages[1]);
+                }
+            }
+          },
+        },
+      ] as MSWMocksHandlersProps['endpointsMocks'],
+    }
+  );
+
+  return { user, ...renderResult };
+}
+
+// TODO @adhitya.mamallan - Explore using fakerjs.dev for cases like this
+function generateWorkflowPages(
```

Comment:
nit: maybe we can store fixtures like a list of workflows in different states, workflows states so that they could be loaded in tests.

Reply:
Yeah, the TODO above alludes to this; there are libraries that handle generating fake data. We'll likely research and pick this up as part of overall testing improvements sometime next year.